Autopilot/Inc/gps.hpp

```diff
@@ -49,51 +58,74 @@ class Gps
         virtual void GetResult(GpsData_t *Data) = 0;
 };
 
-class NEOM8 : public Gps 
+class NEOM8 : public Gps
 {
 	public:
 		NEOM8(const NEOM8*) = delete;
 		static NEOM8* GetInstance();
 
 		/**
-         * Triggers interrupt for new GPS measurement - stores raw data in variables and returns right away
-         * */
+		 * Triggers interrupt for new GPS measurement - stores raw data in variables and returns right away
+		 * */
 		void BeginMeasuring() {}; //No need to call this to get data. Just call GetResult()
 
 		 /**GetResult should:
-         * 1. Reset dataIsNew flag
-         * 2. Transfers raw data from variables to struct
-         * 3. Updates utcTime and status values in struct as well
-         * */
+		 * 1. Reset dataIsNew flag
+		 * 2. Transfers raw data from variables to struct
+		 * 3. Updates utcTime and status values in struct as well
+		 * */
 		void GetResult(GpsData_t *Data);
 
+		/**
+		 * Returns the buffer used to receive GPS UART signals
+		 */
+		uint8_t* get_byte_collection_buffer();
```

Comment:
Do this func and the one below need to be public ? Or can they be private ?

Reply:
They need to be public for the interrupt unfortunately